map/mode.h

```diff
@@ -9,96 +9,53 @@
 #ifndef __MAP_MODE_H__
 #define __MAP_MODE_H__
 
+#include "accelerator/common_core.h"
+#include "accelerator/errors.h"
+#include "common/trinary/flex_trit.h"
+#include "common/trinary/tryte_ascii.h"
 #include "mam/api/api.h"
-
-#define MSS_DEPTH 1
+#include "mam/mam/mam_channel_t_set.h"
+#include "request/request.h"
+#include "response/response.h"
+#include "utarray.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
 /**
- * Creates a channel
+ * Initialize a mam_api_t object
  *
  * @param api - The API [in,out]
- * @param channel_id - A known channel ID [out]
- * @param depth - depth of merkle tree going to generate [in]
+ * @param[in] iconf IOTA API parameter configurations[in]
+ * @param seed - MAM channels seed. It is an optional choice[in]
  *
  * @return return code
  */
-retcode_t map_channel_create(mam_api_t* const api, tryte_t* const channel_id, const size_t depth);
+status_t map_mam_init(mam_api_t* const api, const iota_config_t* const iconf, tryte_t const* const seed,
+                      int32_t channel_ord, mam_psk_t_set_t* const psks, mam_ntru_pk_t_set_t* const ntru_pks,
+                      tryte_t const* const psk, tryte_t const* const ntru_pk);
 
-/**
- * Creates and announce a new channel on header
- *
- * @param api - The API [in,out]
- * @param channel_id - A known channel ID [in]
- * @param bundle - The bundle that the packet will be written into [out]
- * @param msg_id - The msg_id [out]
- * @param new_channel_id - The new channel ID [out]
- *
- * @return return code
- */
-retcode_t map_announce_channel(mam_api_t* const api, tryte_t const* const channel_id,
-                               bundle_transactions_t* const bundle, trit_t* const msg_id,
-                               tryte_t* const new_channel_id);
+status_t map_write_message_to_bundle_on_available_sk(const iota_client_service_t* const service, mam_api_t* const api,
```

Comment:
Can it be renamed to `map_to_bundle_on_avail_sk`?

Reply:
Comments are required.

Reply:
There two kinds of transactions in our MAM design. One is Message and the other one is Announcement. This function can only write Message to bundle. How about I rename it into `map_write_msg_to_bundle()`?

Reply:
And I will add the corresponding comment right away

Reply:
I think the name `map_write_msg` sounds a bit silly since we already know the direction (to write).

Reply:
I want to include `msg` in the function name, since I want to avoid confusing that this function can be used for `Header` which is different form `Message`.

Reply:
How about `map_written_msg_to_bundle_on_available_sk`.

Reply:
I am okay with it. If you think it is to long, the `on_available_sk` string can be removed, since a message must be only written on a available secret key.

Reply:
Let's try `map_written_msg_to_bundle`.

Reply:
OK!